akka-http-core/src/main/scala/akka/http/impl/engine/server/HttpServerBluePrint.scala

```diff
@@ -430,19 +431,24 @@ private[http] object HttpServerBluePrint {
         def onPush(): Unit = {
           val requestStart = openRequests.head
           openRequests = openRequests.tail
+          val onComplete =
+            requestStart.headers.find(_.name.equals(OnComplete.name)).map { case header: OnComplete => header.onCompleteAccess.onComplete }
 
           val response0 = grab(httpResponseIn)
-          val response =
-            if (response0.entity.isStrict) response0 // response stream cannot fail
-            else response0.mapEntity { e =>
-              val (newEntity, fut) = HttpEntity.captureTermination(e)
-              fut.onComplete {
-                case Failure(ex) =>
-                  log.error(ex, s"Response stream for [${requestStart.debugString}] failed with '${ex.getMessage}'. Aborting connection.")
-                case _ => // ignore
-              }(ExecutionContexts.sameThreadExecutionContext)
-              newEntity
-            }
+          val response = response0 mapEntity { e =>
```

Comment:
This looks like it would fairly seriously impact performance - perhaps it should be behind a configuration flag?